### PR TITLE
layers: Early return in cc_pipeline_graphics.cpp

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Some quality-of-life fixes to prevent deep nesting by doing reasonable early returns

broke off in separate PR for ease of review

### CODE REVIEW

File: layers/core_checks/cc_pipeline_graphics.cpp [hunk 1]
```diff
@@ -2484,16 +2479,17 @@ bool CoreChecks::ValidateGraphicsPipelineShaderDynamicState(const vvl::Pipeline
...
-            if (!phys_dev_ext_props.fragment_shading_rate_props.primitiveFragmentShadingRateWithMultipleViewports &&
-                pipeline.IsDynamic(VK_DYNAMIC_STATE_VIEWPORT_WITH_COUNT) && cb_state.dynamic_state_value.viewport_count != 1) {
-                if (stage_state.entrypoint && stage_state.entrypoint->written_builtin_primitive_shading_rate_khr) {
-                    skip |= LogError(vuid.viewport_count_primitive_shading_rate_04552, stage_state.module_state->Handle(), loc,
-                                     "%s shader of currently bound pipeline statically writes to PrimitiveShadingRateKHR built-in"
-                                     "but multiple viewports are set by the last call to vkCmdSetViewportWithCountEXT,"
-                                     "and the primitiveFragmentShadingRateWithMultipleViewports limit is not supported.",
-                                     string_VkShaderStageFlagBits(stage));
-                }
+        if (stage != VK_SHADER_STAGE_VERTEX_BIT && stage != VK_SHADER_STAGE_GEOMETRY_BIT && stage != VK_SHADER_STAGE_MESH_BIT_EXT) {
```
Comment (arno-lunarg) [inline layers/core_checks/cc_pipeline_graphics.cpp:2482]:
```suggestion
        if (!IsValueIn(stage, {VK_SHADER_STAGE_VERTEX_BIT, VK_SHADER_STAGE_GEOMETRY_BIT, VK_SHADER_STAGE_MESH_BIT_EXT})) {
```

**Review by arno-lunarg — APPROVED** [review]
LGTM, cool cleanups!

File: layers/core_checks/cc_pipeline_graphics.cpp [hunk 2]
```diff
@@ -273,6 +273,7 @@ bool CoreChecks::ValidateGraphicsPipelineLibrary(const vvl::Pipeline &pipeline,
                          string_VkShaderStageFlags(pipeline.create_info_shaders).c_str());
     }
 
+    // If extension is not enable, ignore the rest of checks
```
Comment (filnet) [inline layers/core_checks/cc_pipeline_graphics.cpp:276]: typo `enable`.

Comment (spencer-lunarg) [reply layers/core_checks/cc_pipeline_graphics.cpp:276]: fixed, thanks for the catch

**Pull request merged**